src/aiortc/jitterbuffer.py

```diff
@@ -18,35 +19,44 @@ def __init__(self, capacity: int, prefetch: int = 0) -> None:
         self._origin: Optional[int] = None
         self._packets: List[Optional[RtpPacket]] = [None for i in range(capacity)]
         self._prefetch = prefetch
+        self.__max_number = 65536
 
     @property
     def capacity(self) -> int:
         return self._capacity
 
-    def add(self, packet: RtpPacket) -> Optional[JitterFrame]:
+    def add(self, packet: RtpPacket) -> Tuple[bool, Optional[JitterFrame]]:
+        pli_flag = False
         if self._origin is None:
             self._origin = packet.sequence_number
-        elif packet.sequence_number <= self._origin - MAX_MISORDER:
-            self.remove(self.capacity)
-            self._origin = packet.sequence_number
-        elif packet.sequence_number < self._origin:
-            return None
-
-        delta = packet.sequence_number - self._origin
-        if delta >= 2 * self.capacity:
-            # received packet is so far beyond capacity we cannot keep any
-            # previous packets, so reset the buffer
-            self.remove(self.capacity)
-            self._origin = packet.sequence_number
-        elif delta >= self.capacity:
-            # remove just enough packets to fit the received packets
+            delta = 0
+            misorder = 0
+        else:
+            delta = uint16_add(packet.sequence_number, -self._origin)
+            misorder = uint16_add(self._origin, -packet.sequence_number)
+
+        if misorder < delta:
+            if misorder >= MAX_MISORDER:
+                self.remove(self.capacity)
+                self._origin = packet.sequence_number
+                delta = misorder = 0
+                if self._capacity >= 128:
```

Comment:
What's the 128 magic value? Could we have a named constant with a descriptive name?

Reply:
This is the value set in rtcrtpreceiver.py (line 257) as the jitterbuffer size for video stream. I assume that this jitterbuffer size for video will rather increase in time, while the audio buffer should not reach 128 any time soon. I used the jitterbuffer size (capacity) to make sure that we are actually dealing with video before sending PLI.

Reply:
A viable alternative would be to add a "is_video" flag to the jitterbuffer `__init__` method, as currently there is no easy way to check if the jitterbuffer is used for video other then checking the `self._capacity` value.

Reply:
`is_video` sounds like a good idea!